### PR TITLE
Add vendor stickiness test (RhBug:1788371)

### PR DESCRIPTION
**Pull request by inknos**
libdnf: https://github.com/rpm-software-management/libdnf/pull/907
dnf   : https://github.com/rpm-software-management/dnf/pull/1602
BZ    : https://bugzilla.redhat.com/show_bug.cgi?id=1788371

### CODE REVIEW

**Comment by Conan-Kudo** [comment]
@inknos Typo `sticknes` -> `stickiness`

File: dnf-behave-tests/features/stick-to-vendor.feature [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+       | upgrade | vendor-1.1-1.x86_64 |
+  Given I successfully execute rpm with args "-qi vendor"
+   Then the exit code is 0
+   And stdout contains "Vendor      : First Vendor"
+
+@wip
+@bz1788371
+Scenario: No upgrade if same vendor not found
+  Given I use repository "dnf-ci-vendor-2-updates"
+   When I execute dnf with args "upgrade vendor"
```
Comment (dmach) [inline dnf-behave-tests/features/stick-to-vendor.feature]: The command doesn't seem to be using the config file you've created.
Maybe there's a better way of doing it, but adding following option to all upgrade/downgrade commands fixed test failures: ```--config={context.dnf.installroot}/etc/dnf/dnf.conf```

File: dnf-behave-tests/features/stick-to-vendor.feature [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+@bz1788371
+Scenario: No downgrade if same vendor not found
+  Given I use repository "dnf-ci-vendor-1-updates"
+  Given I use repository "dnf-ci-vendor-2"
+   When I execute dnf with args "install vendor-1.1-1"
+  Given I drop repository "dnf-ci-vendor-1"
+   When I execute dnf with args "downgrade vendor"
+   Then the exit code is 0
+   And transaction is empty
```
Comment (dmach) [inline dnf-behave-tests/features/stick-to-vendor.feature]: This ends with:
```Error: 
 Problem: problem with installed package vendor-1.1-1.x86_64
  - cannot install both vendor-1.0-1.x86_64 and vendor-1.1-1.x86_64
  - conflicting requests
```
The error message indicates that it attempts to install both old and new package, possibly with a different vendor.
Please follow up with @j-mracek - he might know what's wrong. I suspect the wrong preselection of downgrade candidates.

**Comment by Conan-Kudo** [comment]
@rh-atomic-bot r+

**Pull request merged**